keps/349-all-or-nothing/README.md

```diff
@@ -226,6 +245,16 @@ type WaitForPodsReady struct {
 	// RequeuingStrategy defines the strategy for requeuing a Workload.
 	// +optional
 	RequeuingStrategy *RequeuingStrategy `json:"requeuingStrategy,omitempty"`
+
+	// RecoveryTimeout defines an optional timeout, measured since the
+	// last transition to the PodsReady=false condition after a Workload is Admitted and running.
+	// Such a transition may happen when a Pod failed and the replacement Pod 
+	// is awaited to be scheduled.
+	// After exceeding the timeout the corresponding job gets suspended again
+	// and requeued after the backoff delay. The timeout is enforced only if waitForPodsReady.enable=true. 
+	// Defaults to 3 mins.
+	// +optional
+	RecoveryTimeout *metav1.Duration `json:"recoveryTimeout,omitempty"`
```

Comment:
As we discussed in the previous meeting, can we disable this and avoid defaulting with

```yaml
recoveryTimeout: {}
```
?

Reply:
I synced with @PBundyra offline. In conclusion, Patryk found that the above approach is technically challenging.

So, he proposed the following solutions:

> - _ If not specified otherwise, .recoveryTimeout is null, and it doesn't impact Kueue's behavior_
> - _Put the 3mins in Kueue's release manifest, then a user that uncomments and enables it, sees that it's. there_
> - _For existing users that use configmap without .recoveryTimeout :_
>     - _Describe the extension in release notes and mark is as important
>     - _Document it well on WaitForPodsReady website in Kueue's documentation_
>     - _Recommend it in documentation for TAS_

Originally posted by @PBundyra